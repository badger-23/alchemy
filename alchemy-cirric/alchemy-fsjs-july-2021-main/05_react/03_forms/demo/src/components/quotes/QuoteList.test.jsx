import React from 'react';
import { render, screen } from '@testing-library/react';
import QuoteList from './QuoteList';

describe('QuoteList', () => {
  it('displays a list of quotes', () => {
    render(
      <QuoteList
        // someObj={{ name: "spot" }}
        quotes={[
          {
            name: 'Bender',
            quote: 'Bite my shiny metal ass.',
            image: 'example.com/image.png',
          },
          {
            name: 'Fry',
            quote:
              // eslint-disable-next-line max-len
              '...it\'s what seperates humans and robots from animals and animal robots.',
            image: 'example.com/image.png',
          },
        ]}
      />
    );

    const ul = screen.getByRole('list', { name: 'quotes' });
    expect(ul).toMatchSnapshot();
  });
});
