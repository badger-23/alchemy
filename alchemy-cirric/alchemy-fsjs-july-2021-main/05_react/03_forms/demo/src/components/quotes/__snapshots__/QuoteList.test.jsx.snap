// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuoteList displays a list of quotes 1`] = `
<ul
  aria-label="quotes"
>
  <li>
    <figure>
      <img
        alt="Bender"
        src="example.com/image.png"
      />
      <figcaption>
        Bite my shiny metal ass.
         - 
        Bender
      </figcaption>
    </figure>
                  
  </li>
  <li>
    <figure>
      <img
        alt="Fry"
        src="example.com/image.png"
      />
      <figcaption>
        ...it's what seperates humans and robots from animals and animal robots.
         - 
        Fry
      </figcaption>
    </figure>
                  
  </li>
</ul>
`;
